packages/block-editor/src/components/inserter/menu.js

```diff
@@ -251,6 +251,12 @@ export class InserterMenu extends Component {
 			! isEmpty( itemsPerCollection );
 		const hoveredItemBlockType = hoveredItem ? getBlockType( hoveredItem.name ) : null;
 		const hasHelpPanel = hasItems && showInserterHelpPanel;
+		const viewportWidth =
```

Comment:
@scruffian Neat! Prop pass through with value fallback :)

I lack a bit of project context for this update. In isoloation, the change makes sense to me.

For implementation, I feel like we can clean up the `&&` checks by using [lodash.get](https://lodash.com/docs/4.17.15#get) (since the package already has `lodash` as a dependency)

That way, you can do something like this:

```js
import { get } from 'lodash'

const viewportWidth = get(hoveredItemBlockType, 'example.viewportWidth', 500)
```

Would love your feedback 😊

Reply:
Great idea thanks! Done in ec596e0fb.

Reply:
@ItsJonQ Curious, does gutenberg support the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) (meaning, is it transpiled in the build tools for browser compatibility)?

Reply:
@creativecoder Good question! I remember this was mentioned recently. It looks like something was merged https://github.com/WordPress/gutenberg/pull/19831

I haven't personally tried it yet. @gziolo Is this project optional chaining ready? :D

Reply:
It should support, however in practice there was a blocker discovered by @aduth and filed in https://github.com/WordPress/gutenberg/issues/19952. `docgen` tool that we use for documentation doesn't support ES2019 and ES2020 😞